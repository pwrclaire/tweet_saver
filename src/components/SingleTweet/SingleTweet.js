import React from "react";
import styled from "styled-components";
import moment from "moment";

const TweetContainer = styled.div`
  background-color: white;
  padding: 5px;
  margin-bottom: 13px;
  width: 100%;
  display: flex;
  cursor: ${(props) => (props.draggable ? "pointer" : "")};
  box-shadow: 3px 3px 10px 0px rgba(0, 0, 0, 0.75);
  box-sizing: border-box;
`;

const AvatarContainer = styled.div`
  width: 50px;
  height: 50px;
  padding: 5px;
`;

const InfoContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

const UserInfo = styled.div`
  display: flex;
`;

const SingleTweet = ({ tweet, canDrag, ondrag }) => {
  const { text, user } = tweet;
  const { name, screenName, profileImageURLHttps } = user;
  const date = moment(tweet.createdAt).format("MMM D, YYYY");
  return (
    <TweetContainer
      onDragStart={(e) => ondrag(e)}
      draggable={canDrag}
      id={tweet.id}
    >
      <AvatarContainer>
        <img src={profileImageURLHttps} alt="" />
      </AvatarContainer>
      <InfoContainer>
        <UserInfo>
          <b>{name}</b>&nbsp;@{screenName}
          {"  "}
          {date}
        </UserInfo>
        {text}
      </InfoContainer>
    </TweetContainer>
  );
};

export { SingleTweet };
