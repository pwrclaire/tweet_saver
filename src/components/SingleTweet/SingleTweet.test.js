import React from "react";
import { SingleTweet } from "./SingleTweet";
import renderer from "react-test-renderer";

describe("Testing SingleTweet component", () => {
  test("should render", () => {
    const component = renderer.create(<SingleTweet tweet={mockData} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const mockData = {
  createdAt: 1591476978000,
  id: 1269372565350494200,
  text: "@Friendlyperson0 Test test",
  user: {
    name: "GjeniAzelia",
    screenName: "AzeliaGjeni",
    profileImageUrlHttps:
      "https://pbs.twimg.com/profile_images/1225850319516635138/AAQ-gnQy_normal.jpg",
  },
};
