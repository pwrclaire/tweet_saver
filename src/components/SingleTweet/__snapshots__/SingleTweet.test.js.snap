// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SingleTweet component should render 1`] = `
<div
  className="sc-AxjAm fGihqS"
  id={1269372565350494200}
  onDragStart={[Function]}
>
  <div
    className="sc-AxirZ cXrbly"
  >
    <img
      alt=""
    />
  </div>
  <div
    className="sc-AxiKw jWoTjM"
  >
    <div
      className="sc-AxhCb ZaNiK"
    >
      <b>
        GjeniAzelia
      </b>
      Â @
      AzeliaGjeni
        
      Jun 6, 2020
    </div>
    @Friendlyperson0 Test test
  </div>
</div>
`;
